systemtest/agentconfig_test.go

```diff
@@ -117,6 +119,15 @@ func queryAgentConfig(t testing.TB, serverURL, serviceName, serviceEnvironment,
 		req.Header.Set("If-None-Match", etag)
 	}
 	resp, err := http.DefaultClient.Do(req)
+
+	maxRetries := 10
+	var retries int
+	for errors.Is(err, io.EOF) && retries < maxRetries {
+		retries++
+		t.Logf("fleet returned EOF on read, retry %d/%d...", retries, maxRetries)
```

Comment:
```suggestion
		t.Logf("apm-server returned EOF on read, retry %d/%d...", retries, maxRetries)
```

These queries are going to apm-server's APM agent config endpoint.

Reply:
sorry, I had misread the test and thought we were querying the Agents endpoint on Fleet. I agree, that this is not a great fix. I had only resorted to a retry since I thought it was an external dependency that wasn't coming up in time. I've opened an issue to determine the root cause of this and fix it, instead of relying on the retry.